student-preferred-name-portlet-webapp/src/main/java/edu/wisc/portlet/preferred/service/PreferredNameServiceImpl.java

```diff
@@ -59,16 +59,20 @@ public String getStatus(PreferredName ldapPn) {
             && (jdbcPn == null
             || (StringUtils.isEmpty(jdbcPn.getFirstName()) && StringUtils
             .isEmpty(jdbcPn.getMiddleName()) && StringUtils.isEmpty(jdbcPn.getLastName()))
-        )
-            ) {
+            )){
             return "(not set)";
-        } else {
-
-            if (ldapPn.equals(jdbcPn)) {
+        }else if(jdbcPn == null || 
+                (StringUtils.isEmpty(jdbcPn.getFirstName()) && 
+                 StringUtils.isEmpty(jdbcPn.getMiddleName()) && 
+                 StringUtils.isEmpty(jdbcPn.getLastName()))){ 
+            return "(deletion pending)";
+        }else{
+            String ldapPreferredName = new StringBuilder().append(ldapPn.getFirstName()).append(ldapPn.getMiddleName()).toString();
+            String jdbcPreferredName = new StringBuilder().append(jdbcPn.getFirstName()).append(jdbcPn.getMiddleName()).toString();
+            if(ldapPreferredName.equals(jdbcPreferredName)){ 
```

Comment:
This logic is not detecting when a change to only the last name is pending -- that edge case will be experienced as no change pending.  Backlogged as [SPNPTL-102](https://jira.doit.wisc.edu/jira/browse/SPNPTL-102).

May be worth an inline code comment to this effect to guide the reader in what this logic is doing and in particular why it is not considering last name.